### ISSUE TITLE
Race condition in metrics emitter

### ISSUE THREAD

**Issue by jomsie** [concourse/concourse#4310]
## Bug Report

When running `gingko -race` in the atc/metric folder we see a race condition.

## Steps to Reproduce

`cd` into `atc/metric`
run `gingko -race`

## Expected Results

... no race condition

## Actual Results

```
WARNING: DATA RACE
Write at 0x00000127b390 by goroutine 17:
  github.com/concourse/concourse/atc/metric.Initialize()
      /tmp/build/0df9eea0/concourse/atc/metric/emit.go:106 +0x4d6
```

https://ci.concourse-ci.org/teams/main/pipelines/concourse/jobs/unit/builds/1027#L5d5aa9e3:808:978

## Additional Context

Might be something in here? https://golang.org/doc/articles/race_detector.html#Typical_Data_Races

## Version Info

* Concourse version: v5.5.0
* Deployment type (BOSH/Docker/binary): bosh
* Infrastructure/IaaS: GCP
* Browser (if applicable): Chrome
* Did this used to work? It seems to have been an issue for a while, but only if you use the `-race` flag, but we've only seen it on prod (unit) once so far.

**Comment by jomsie** [comment]
Denise sayz:

There probably is an issue with how `emissions` is synchronously declared outside of the goroutine, but also being manipulated and used within the goroutine. The code kind of resembles the example here: https://www.sohamkamani.com/blog/2018/02/18/golang-data-race-and-how-to-fix-it/

**Comment by deniseyu** [comment]
We can make the data race go away by wrapping both locations where writes are happening with mutexes:

concourse/atc/metric/emit.go:107
```
	mutex.Lock()
	emissions = make(chan eventEmission, int(bufferSize))
	mutex.Unlock()
```

concourse/atc/metric/emit.go:149
```
func emitLoop() {
	mutex.Lock()
	defer mutex.Unlock()

	for emission := range emissions {
		emitter.Emit(emission.logger.Session("emit"), emission.event)
	}
}
```

but this might not be the best way to address race conditions in this part of the code - we will have to test to see whether this introduces latency in metric emissions.

**Comment by pivotal-jamie-klassen** [comment]
uh, i think we fixed this in #4496. @deniseyu does that seem right to you?

**Comment by deniseyu** [comment]
@pivotal-jamie-klassen the data race error message suggests that there was some issue in the source code, but if we're not seeing the error anymore then happy to call it `fixed`! I'll close this out.
